packages/sds/src/message_channel/repair/buffers.ts

```diff
@@ -0,0 +1,280 @@
+import { Logger } from "@waku/utils";
+
+import type { HistoryEntry } from "../message.js";
+
+const log = new Logger("sds:repair:buffers");
+
+/**
+ * Entry in the outgoing repair buffer with request timing
+ */
+interface OutgoingBufferEntry {
+  entry: HistoryEntry;
+  tReq: number; // Timestamp when this repair request should be sent
+  requested: boolean; // Whether this repair has been requested already
+}
+
+/**
+ * Entry in the incoming repair buffer with response timing
+ */
+interface IncomingBufferEntry {
+  entry: HistoryEntry;
+  tResp: number; // Timestamp when we should respond with this repair
+}
+
+/**
+ * Buffer for outgoing repair requests (messages we need)
+ * Maintains a sorted array by T_req for efficient retrieval of eligible entries
+ */
+export class OutgoingRepairBuffer {
+  // Sorted array by T_req (ascending - earliest first)
+  private items: OutgoingBufferEntry[] = [];
+  private readonly maxSize: number;
+
+  public constructor(maxSize = 1000) {
+    this.maxSize = maxSize;
+  }
+
+  /**
+   * Add a missing message to the outgoing repair request buffer
+   * If message already exists, it is not updated (keeps original T_req)
+   * @returns true if the entry was added, false if it already existed
+   */
+  public add(entry: HistoryEntry, tReq: number): boolean {
+    const messageId = entry.messageId;
+
+    // Check if already exists - do NOT update T_req per spec
+    const existingIndex = this.items.findIndex(
+      (item) => item.entry.messageId === messageId
+    );
+    if (existingIndex !== -1) {
+      log.info(
+        `Message ${messageId} already in outgoing buffer, keeping original T_req`
+      );
+      return false;
+    }
+
+    // Check buffer size limit
+    if (this.items.length >= this.maxSize) {
+      // Evict furthest T_req entry (last in sorted array) to preserve most urgent repairs
+      const evicted = this.items.pop()!;
+      log.warn(
+        `Buffer full, evicted furthest entry ${evicted.entry.messageId} with T_req ${evicted.tReq}`
+      );
+    }
+
+    // Add new entry and re-sort
+    const newEntry: OutgoingBufferEntry = { entry, tReq, requested: false };
+    const combined = [...this.items, newEntry];
+
+    // Sort by T_req (ascending)
+    combined.sort((a, b) => a.tReq - b.tReq);
+
+    this.items = combined;
+    log.info(`Added ${messageId} to outgoing buffer with T_req: ${tReq}`);
+    return true;
+  }
+
+  /**
+   * Remove a message from the buffer (e.g., when received)
+   */
+  public remove(messageId: string): void {
+    this.items = this.items.filter(
+      (item) => item.entry.messageId !== messageId
+    );
+  }
+
+  /**
+   * Get eligible repair requests (where T_req <= currentTime)
+   * Returns up to maxRequests entries from the front of the sorted array
+   * Marks returned entries as requested but keeps them in buffer until received
+   */
+  public getEligible(currentTime: number, maxRequests = 3): HistoryEntry[] {
```

Comment:
```suggestion
  public getEligible(currentTime: number = Date.now(), maxRequests = 3): HistoryEntry[] {
```

I think that's fine to do this. Makes it still testable.